lib/features/identity_creator/presentation/identity_creator_controller.dart

```diff
@@ -160,21 +168,12 @@ class IdentityCreatorController extends BaseController {
   }
 
   void _getAllIdentities() {
-    log('IdentityCreatorController::_getAllIdentities() ');
-    if (accountId != null) {
-      try {
-        requireCapability(session!, accountId!, [CapabilityIdentifier.jamesSortOrder]);
-        consumeState(_getAllIdentitiesInteractor.execute(
-          accountId!,
-          properties: Properties({'email', 'sortOrder'})
-        ));
-      } catch (e) {
-        logError('IdentityCreatorController::_getAllIdentities(): exception: $e');
-        consumeState(_getAllIdentitiesInteractor.execute(
-          accountId!,
-          properties: Properties({'email'})
-        ));
-      }
+    if (accountId != null && session != null) {
+      final propertiesRequired = [CapabilityIdentifier.jamesSortOrder].isSupported(session!, accountId!)
```

Comment:
If written like you it's not generic. We just need to write a function `isSupported` to check the list of supported `capability`. It can be used in many different situations, not necessarily just `default identity`